module-iam.tf

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  create_iam_worker_policy = anytrue([
+    var.create_iam_worker_policy == "always",
+    var.create_iam_worker_policy == "auto" && var.create_cluster && anytrue([for k, v in var.worker_pools :
+      tobool(lookup(v, "enabled", var.worker_pool_enabled)) &&
+      lookup(v, "mode", var.worker_pool_mode) != "node-pool"
+    ])
+  ])
+
+  create_iam_autoscaler_policy = anytrue([
+    var.create_iam_autoscaler_policy == "always",
+    var.create_iam_autoscaler_policy == "auto" && var.create_cluster && anytrue([for k, v in var.worker_pools :
+      tobool(lookup(v, "enabled", var.worker_pool_enabled)) &&
+      tobool(lookup(v, "allow_autoscaler", false))
+    ])
+  ])
+
+  create_iam_operator_policy = anytrue([
+    var.create_iam_operator_policy == "always",
+    var.create_iam_operator_policy == "auto" && local.operator_enabled
+  ])
+
+  create_iam_kms_policy = anytrue([
+    var.create_iam_kms_policy == "always",
+    var.create_iam_kms_policy == "auto" && anytrue([
+      coalesce(var.worker_volume_kms_key_id, "none") != "none",
+      coalesce(var.cluster_kms_key_id, "none") != "none",
+    ])
+  ])
+}
+
+# Default IAM sub-module implementation for OKE cluster
+module "iam" {
+  source                       = "./modules/iam"
+  compartment_id               = local.compartment_id
+  state_id                     = random_id.state_id.id
+  tenancy_id                   = local.tenancy_id
+  cluster_id                   = local.cluster_id
+  create_iam_resources         = var.create_iam_resources
+  create_iam_autoscaler_policy = local.create_iam_autoscaler_policy
+  create_iam_kms_policy        = local.create_iam_kms_policy
+  create_iam_operator_policy   = local.create_iam_operator_policy
+  create_iam_worker_policy     = local.create_iam_worker_policy
+
+  create_iam_tag_namespace = var.create_iam_tag_namespace
+  create_iam_defined_tags  = var.create_iam_defined_tags
+  defined_tags             = lookup(var.defined_tags, "policy", {})
+  freeform_tags            = lookup(var.freeform_tags, "policy", {})
+  tag_namespace            = var.tag_namespace
+  use_defined_tags         = var.use_defined_tags
+
+  cluster_kms_key_id         = var.cluster_kms_key_id
+  operator_volume_kms_key_id = var.operator_volume_kms_key_id
+  worker_volume_kms_key_id   = var.worker_volume_kms_key_id
+
+  autoscaler_compartments = local.autoscaler_compartments
+  worker_compartments     = local.worker_compartments
+
+  providers = {
+    oci.home = oci.home
+  }
+}
+
+moved {
+  from = module.policy
+  to   = module.iam
+}
+
+output "availability_domains" {
```

Comment:
Similar to locals, I think it would be good to group the outputs in 1 file. In this example, the `local.ad_numbers_to_names` is defined in data-common.tf but the output is here. 

Let's put all the outputs in 1 file and document the sections. Outputs should be either limited to just enough that makes the module reusable and allow other users to build on this module directly or to find more details using the API if they need to.

Reply:
Point taken on these definitions belonging to `iam` and corrected. We can discuss further in the other related thread.